app/models/authorization.rb

```diff
@@ -0,0 +1,30 @@
+class Authorization < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :o_auth_application
+  has_many :scopes, through: :authorization_scopes
+  has_many :token
+
+  before_validation :setup, on: :create
+
+  validates :refresh_token, presence: true, uniqueness: true
```

Comment:
Validating uniqueness of the (user, o_auth_application) tuple might be good too.